infra/storage/file_transfer_managers/s3Mgr.class.php

```diff
@@ -222,22 +222,16 @@ protected function doChmod ($remote_file, $chmod_code)
 	// return true/false according to existence of file on the server
 	protected function doFileExists($remote_file)
 	{
+		$this->s3->registerStreamWrapper();
+
 		list($bucket, $remote_file) = explode("/",ltrim($remote_file,"/"),2);
-		if($this->isdirectory($remote_file)) 
-		{
-			return true;
-		}
-		KalturaLog::debug("remote_file: ".$remote_file);
+		$s3path = "s3://" . $bucket . '/' . $remote_file;
 
-		$exists = $this->s3->doesObjectExist($bucket, $remote_file);
-		return $exists;
-	}
+		KalturaLog::debug("remote_file: ".$s3path);
 
-	private function isdirectory($file_name) {
-		if(strpos($file_name,'.') === false) return TRUE;
-		return false;
+		return is_dir($s3path) || file_exists($s3path);
```

Comment:
This specific change needs to be revised, 
Since is_dir & file_exists will check actual path on local file system , while isdirectory  just checked that it is a valid dir name and doesObjectExist validated remote object.